packages/clay-form/src/DualListBox.tsx

```diff
@@ -0,0 +1,182 @@
+
+function swapArrayItems(arrays: TItems, selectedIndexes: Array<number>) {
+	const [sourceArray, targetArray] = arrays;
+
+	const newTargetArray = [...targetArray];
+
+	const newSourceArray = sourceArray.filter((item, index) => {
+		if (selectedIndexes.includes(index)) {
+			newTargetArray.push(item);
+
+			return false;
+		}
+
+		return true;
+	});
+
+	return [newSourceArray, newTargetArray];
+}
+
+interface IBoxProps {
+	/**
+	 * Label of the left Select Box.
+	 */
+	label?: string;
+
+	/**
+	 * Handler that triggers when a new item is selected in the left Select Box.
+	 */
+	onSelectChange?: (val: Array<string>) => void;
+
+	/**
+	 * Array of selected items in the left Select Box.
+	 */
+	selected?: Array<string>;
+}
+
+interface IProps extends React.HTMLAttributes<HTMLDivElement> {
+	/**
+	 * Items spread across two arrays that will be displayed in the two Select Boxes.
+	 */
+	items: TItems;
+
+	/**
+	 * Handler that triggers when the content of the items prop are changed. Caused by either reordering or transfering of items.
+	 */
+	onItemsChange: (val: TItems) => void;
+
+	/**
+	 * Props for the left Select Box.
+	 */
+	left?: IBoxProps;
+
+	/**
+	 * Props for the right Select Box.
+	 */
+	right?: IBoxProps;
+
+	/**
+	 * Amount of items that can fit inside the both Select Boxes before a scrollbar is introduced.
+	 */
+	size?: number;
+
+	/**
+	 * Path to the spritemap that Icon should use when referencing symbols.
+	 */
+	spritemap?: string;
+}
+
+const ClayDualListBox: React.FunctionComponent<IProps> = ({
+	className,
+	items,
+	left = {},
+	onItemsChange,
+	right = {},
+	size,
+	spritemap,
+	...otherProps
+}) => {
+	const [internalLeftSelected, setInternalLeftSelected] = React.useState(
+		left.selected || []
+	);
+	const [internalRightSelected, setInternalRightSelected] = React.useState(
+		right.selected || []
+	);
+
+	const handleLeftSelectedChange =
+		left.onSelectChange || setInternalLeftSelected;
+	const handleRightSelectedChange =
+		right.onSelectChange || setInternalRightSelected;
+
+	const leftSelected = left.selected || internalLeftSelected;
+	const rightSelected = right.selected || internalRightSelected;
+
+	const [leftItems, rightItems] = items;
+
+	const selectedIndexesLeft = getSelectedIndexes(leftItems, leftSelected);
+	const selectedIndexesRight = getSelectedIndexes(rightItems, rightSelected);
+
+	return (
+		<div {...otherProps} className={classNames(className, 'form-group')}>
+			<div className="clay-dual-listbox">
+				<ClaySelectBox
+					className="clay-dual-listbox-item clay-dual-listbox-item-expand listbox-left"
+					items={leftItems}
+					label={left.label}
+					multiple
+					onItemsChange={newLeftItems =>
+						onItemsChange([newLeftItems, rightItems])
+					}
+					onSelectChange={handleLeftSelectedChange}
+					showArrows
+					size={size}
+					spritemap={spritemap}
+					value={leftSelected}
+				/>
+
+				<ClayButton.Group className="clay-dual-listbox-actions clay-dual-listbox-item">
```

Comment:
Can we add at `type="vertical"` or `type="stacked` and have the container be `<div class="btn-group-vertical">` instead of `<div class="btn-group btn-group-vertical">`? If not I can add a css overwrite for this one.

Reply:
Hey @pat270, not exactly sure what you mean by that suggestion, but it stroke me as odd... I'm not sure `type` is a valid `html` attribute for a `div` tag. If we need some kind of specifier, we can always use `data-type`, though.

Reply:
`type` doesn't work on `ClayButton.Group` or `div`. Replacing it with `<div class="btn-group-vertical">` makes them squished.

Reply:
@jbalsas doesn't need to be type just something to make it vertical. I guess you could even get rid of `ClayButton.Group` here and just use a `<div  className="btn-group-vertical clay-dual-listbox-actions clay-dual-listbox-item">`.

Reply:
@pat270 They are vertical now as it stands.

Reply:
@kresimir-coko `btn-group-vertical` and `btn-sm` on `ClayButtonWithIcon` will make them the correct size and style.  They are vertical now as it stands is no good because the styles are off.

Regular sized `btn-monospaced` gets squished because Bootstraps `.btn-group-vertical > .btn` styles are overwriting `btn-monospaced`. I need to fix this but not in this PR.

Reply:
Alright @pat270 , I'll add the proper container in the updated PR :)